appgate/syncer/git.py

```diff
@@ -119,8 +119,8 @@ def user_fork(self) -> str | None:
     def checkout_branch(
         self,
         previous_branch: str | None,
-        previous_pr: PullRequest | ProjectMergeRequest | None,
-    ) -> Tuple[str, PullRequest | ProjectMergeRequest | None]:
+        previous_pr: int | None,
+    ) -> Tuple[str, int | None]:
```

Comment:
Is that because of typing? Because you could use a TypeVar instead, like this:
```Python
PullRequestLikeT = TypeVar("PullRequestLikeT", bound=PullRequestLike)

def checkout_branch(
    self,
    previous_pr: PullRequestLikeT
) -> PullRequestLikeT:
```

The optional makes it a bit more annoying though.

Reply:
If sticking to an int you could use a NewType to make sure we pass the right thing around:
```Python
from typing import NewType

PullRequestNumber = NewType("PullRequestNumber", int)
```

That's only used a type-check time, at runtime it would still be an int.

Reply:
I think we should return there the generic prototype for things that look like a pullrequest/mergerequest, that was the idea of having a protocol